guardian-ui/src/GuardianApi.ts

```diff
@@ -0,0 +1,236 @@
+  getVerifyConfigHash: () => Promise<PeerHashMap>;
+  runDkg: () => Promise<void>;
+  startConsensus: () => Promise<void>;
+
+  // Running RPC methods (only exist after run_consensus)
+  version: () => Promise<Versions>;
+  fetchEpochCount: () => Promise<number>;
+  consensusStatus: () => Promise<ConsensusStatus>;
+}
+
+const SESSION_STORAGE_KEY = 'guardian-ui-key';
+
+export class GuardianApi implements ApiInterface {
+  private websocket: JsonRpcWebsocket | null = null;
+  private connectPromise: Promise<JsonRpcWebsocket> | null = null;
+
+  /*** WebSocket methods ***/
+
+  public connect = async (): Promise<JsonRpcWebsocket> => {
+    if (this.websocket !== null) {
+      return this.websocket;
+    }
+    if (this.connectPromise) {
+      return await this.connectPromise;
+    }
+
+    this.connectPromise = new Promise((resolve, reject) => {
+      const websocketUrl = process.env.REACT_APP_FM_CONFIG_API;
+
+      if (!websocketUrl) {
+        throw new Error('REACT_APP_FM_CONFIG_API not set');
+      }
+
+      const requestTimeoutMs = 20000;
+      const websocket = new JsonRpcWebsocket(
+        websocketUrl,
+        requestTimeoutMs,
+        (error: JsonRpcError) => {
+          console.error('failed to create websocket', error);
+          reject(error);
+          this.shutdown();
+        }
+      );
+      websocket
+        .open()
+        .then(() => {
+          this.websocket = websocket;
+          resolve(this.websocket);
+        })
+        .catch((error) => {
+          console.error('failed to open websocket', error);
+          reject(
+            new Error(
+              'Failed to connect to API, confirm your server is online and try again.'
+            )
+          );
+        });
+    });
+
+    return this.connectPromise;
+  };
+
+  shutdown = async (): Promise<boolean> => {
+    if (this.websocket) {
+      const evt: CloseEvent = await this.websocket.close();
+      this.websocket = null;
+      return evt.type === 'close' && evt.wasClean;
+    }
+
+    return true;
+  };
+
+  getPassword = (): string | null => {
+    return sessionStorage.getItem(SESSION_STORAGE_KEY);
+  };
+
+  testPassword = async (password: string): Promise<boolean> => {
+    // Replace with password to check.
+    sessionStorage.setItem(SESSION_STORAGE_KEY, password);
+
+    // Attempt a "status" rpc call with the temporary password.
+    try {
+      await this.status();
+      return true;
+    } catch (err) {
+      // TODO: make sure error is auth error, not unrelated
+      this.clearPassword();
+      return false;
+    }
+  };
+
+  /*** Shared RPC methods */
+
+  status = async (): Promise<ServerStatus> => {
+    const statusResponse: StatusResponse = await this.rpc('status');
+    return statusResponse.server;
+  };
+
+  /*** Setup RPC methods ***/
+
+  setPassword = async (password: string): Promise<void> => {
+    sessionStorage.setItem(SESSION_STORAGE_KEY, password);
+
+    return this.rpc('set_password');
```

Comment:
Nah, we send auth with the request so since it's included, it seems the backend just pulls it from that instead of a param. Very counter-intuitive, could use a comment.